### PR TITLE
watcher: add version 0.9.2, remove older versions

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **watcher/***

- watcher support C++17 since 0.9.0
- API is changed in 0.9.0

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/watcher/all/patches/0.9.0-include-atomic.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/include/wtr/watcher.hpp b/include/wtr/watcher.hpp
+index 7fa556c..9e53bca 100644
+--- a/include/wtr/watcher.hpp
++++ b/include/wtr/watcher.hpp
+@@ -1247,6 +1247,7 @@ inline auto watch(
+ 
+ #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 7, 0)) || defined(__ANDROID_API__)
+ 
++#include <atomic>
```
Comment (RubenRBS) [inline recipes/watcher/all/patches/0.9.0-include-atomic.patch:9]: Has this been reported upstream?

**Review by RubenRBS — COMMENTED** [review]
Thanks! A question about the new patch :)

File: recipes/watcher/all/patches/0.9.0-include-atomic.patch (hunk 1 above)
Comment (toge) [reply recipes/watcher/all/patches/0.9.0-include-atomic.patch:9]: @RubenRBS 
Not yet.
I found several problems about new release.
I am checking these.

**Comment by e-dant** [comment]
Hi all -- Thank you again for you're work on this project

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**